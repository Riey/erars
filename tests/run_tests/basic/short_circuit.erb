@SYSTEM_TITLE
IF 0 && SIDE_EFFECT_NOT()
ELSEIF 0 & SIDE_EFFECT_OK()
ENDIF

IF !(1 || SIDE_EFFECT_NOT())
ELSEIF !(1 | SIDE_EFFECT_OK())
ENDIF

@SIDE_EFFECT_NOT
#FUNCTION
THROW Shouldn't be called

@SIDE_EFFECT_OK
#FUNCTION
PRINTL Should be called
RETURNF 1
