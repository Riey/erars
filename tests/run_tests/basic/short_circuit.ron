[
    Print("Should be called"),
    NewLine,
    Print("Should be called"),
    NewLine,
]
