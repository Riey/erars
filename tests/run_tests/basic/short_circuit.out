Should be called
Should be called
